include/zephyr/net/coap.h

```diff
@@ -1112,6 +1112,15 @@ void coap_pending_clear(struct coap_pending *pending);
  */
 void coap_pendings_clear(struct coap_pending *pendings, size_t len);
 
+/**
+ * @brief Count number of pending requests.
+ *
+ * @param len Number of elements in array.
+ * @param pendings Array of pending requests.
+ * @return count of elements where timeout is not zero.
+ */
+size_t coap_pendings_count(size_t len, struct coap_pending pendings[static len]);
```

Comment:
I'd use a pointer instead (`struct coap_pending *pendings`) just as other pending APIs do. Plus maybe make `len` the second paramter, to align with other APIs.

Reply:
This would have been proper C99 and would not allow NULL pointers.

However, looks like FFF is not able to fake this function.
My old GCC accept `DECLARE_FAKE_VALUE_FUNC(size_t, coap_pendings_count, size_t, struct coap_pending *)` but the new GCC which is used in the CI builds seem to not like it.

So, I'll change the API to use plain pointer and make it match what the other CoAP APIs look like.